alphastats/gui/utils/preprocessing_helper.py

```diff
@@ -1,124 +1,225 @@
-                subset=subset,
-                normalization=normalization,
-                imputation=imputation,
-            )
-            preprocessing = st.session_state.dataset.preprocessing_info
-            st.info(
-                "Data has been processed. "
-                + datetime.datetime.now().strftime("%d/%m/%Y %H:%M:%S")
-            )
-            st.dataframe(
-                pd.DataFrame.from_dict(preprocessing, orient="index").astype(str),
-                use_container_width=True,
-            )
-
-        st.markdown("#### Batch correction: correct for technical bias")
-
-        with st.form("Batch correction: correct for technical bias"):
-            batch = st.selectbox(
-                "Batch", options=st.session_state.dataset.metadata.columns.to_list()
-            )
-            submit_batch_correction = st.form_submit_button("Submit")
-
-        if submit_batch_correction:
-            st.session_state.dataset.batch_correction(batch=batch)
-            st.info(
-                "Data has been processed. "
-                + datetime.datetime.now().strftime("%d/%m/%Y %H:%M:%S")
-            )
-
-    with c2:
-        if submitted:
-            st.markdown("**Intensity Distribution after preprocessing per sample**")
-            fig_processed = st.session_state.dataset.plot_sampledistribution()
-            st.plotly_chart(
-                fig_processed.update_layout(plot_bgcolor="white"),
-                use_container_width=True,
-            )
-
-        else:
-            st.markdown("**Intensity Distribution per sample**")
-            fig_none_processed = st.session_state.dataset.plot_sampledistribution()
-            st.plotly_chart(
-                fig_none_processed.update_layout(plot_bgcolor="white"),
-                use_container_width=True,
-            )
-
-    reset_steps = st.button("Reset all Preprocessing steps")
-
-    if reset_steps:
-        reset_preprocessing()
+    imputation = st.selectbox(
+        "Imputation", options=[None, "mean", "median", "knn", "randomforest"]
+    )
+
+    batch = st.selectbox(
+        "Batch",
+        options=[False] + dataset.metadata.columns.to_list(),
+    )
+
+    return {
+        "remove_contaminations": remove_contaminations,
+        "remove_samples": remove_samples,
+        "subset": subset,
+        "data_completeness": data_completeness,
+        "log2_transform": log2_transform,
+        "normalization": normalization,
+        "imputation": imputation,
+        "batch": batch,
+    }
+
+
+def update_workflow(
+    settings: dict,
+):
+    new_workflow = [
+        key
+        for key, setting in settings.items()
+        if bool(setting)  # use of falsiness
+    ]
+    return new_workflow
+
+
+def run_preprocessing(
+    settings,
+):
+    settings["remove_samples"] = (
+        settings["remove_samples"] if len(settings["remove_samples"]) != 0 else None
+    )
+    st.session_state.dataset.preprocess(**settings)
```

Comment:
could you add a TODO here to pass the whole `settings` object and introduce string constants to access it? (this way it's a bit fragile as the parameters of the `preprocess` method need to be compatible with some strings defined completely elsewhere (`configure_preprocessing()`)